quiche/src/stream/mod.rs

```diff
@@ -816,44 +816,41 @@ impl PartialEq for StreamPriorityKey {
 impl Eq for StreamPriorityKey {}
 
 impl PartialOrd for StreamPriorityKey {
-    // Priority ordering is complex, disable Clippy warning.
-    #[allow(clippy::non_canonical_partial_ord_impl)]
     fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for StreamPriorityKey {
+    fn cmp(&self, other: &Self) -> cmp::Ordering {
         // Ignore priority if ID matches.
         if self.id == other.id {
-            return Some(cmp::Ordering::Equal);
+            return cmp::Ordering::Equal;
```

Comment:
I'm worried about correctness due to this line.

Ignoring the other fields if the ids are equal seems very very bad as it doesn't allow for correct sorting for a collection of elements when there are 2 with same id but differences in other fields.